atom/browser/ui/file_dialog_mac.mm

```diff
@@ -112,7 +112,8 @@ void SetupDialogForProperties(NSOpenPanel* dialog, int properties) {
 // Run modal dialog with parent window and return user's choice.
 int RunModalDialog(NSSavePanel* dialog, atom::NativeWindow* parent_window) {
   __block int chosen = NSFileHandlingPanelCancelButton;
-  if (!parent_window || !parent_window->GetNativeWindow()) {
+  if (!parent_window || !parent_window->GetNativeWindow() ||
+      parent_window->IsOffScreenDummy()) {
```

Comment:
The dialog API should not need to know whether a window is offscreen, this is the responsibility of the caller.

You can pass the offscreen state to `CommonWebContentsDelegate` and `WebDialogHelper` and let those classes decide how to show dialogs.